### PR TITLE
NO-JIRA: introduce JSON schema & validation for rapid container logs

### PR DESCRIPTION
**Pull request by tremes**
This PR implements a new data enhancement to...

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [X] Feature
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

- no new data

## Documentation

- 

## Unit Tests

- 

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

Yes/No

## References

https://issues.redhat.com/browse/???
https://access.redhat.com/solutions/???

### CODE REVIEW

File: pkg/gatherers/conditional/container_log.schema.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            "pattern": "^openshift-[a-zA-Z0-9_.-]{1,128}$|^kube-[a-zA-Z0-9_.-]{1,128}$"
+        },
+        "pod_name_regex": {
+            "type": "string",
+            "description": "Regular expression used to find matching Pods"
+        },
+        "messages": {
+            "type": "array",
+            "description": "The list of regular expressions used to filter the container logs",
+            "uniqueItems": true,
```
Comment (tremes) [inline pkg/gatherers/conditional/container_log.schema.json:34]: @jholecek-rh can/should we do this? i.e require regular expressions to be unique in one particular "container log request".

Comment (jholecek-rh) [reply pkg/gatherers/conditional/container_log.schema.json:34]: What would we seek to achieve by requiring unique regexes?

1. Smaller remote config? I think there are much more significant factors that affect the size. 
2. Faster regex execution/ parsing? The Insights Operator can deduplicate the regexes before combining them into the single one used for parsing, but I think the performance gains would be negligible if any at all (I don't expect duplicates, deduplication would take a little time).

If I compare this with the potential complexity of making sure that there are no duplicates (along with timely feedback to data requestors), I wouldn't bother and allow duplicates.

Comment (tremes) [reply pkg/gatherers/conditional/container_log.schema.json:34]: Mainly the point number 1. The deduplication is in place. OK so let's remove this validation requirement. Thanks

**Review by ncaak — APPROVED** [review]
/lgtm

**Comment by tremes** [comment]
This is not such complex change and techpreview job is failing probably because cloud-credential is marked as degraded for some reason. 
/override ci/prow/e2e-gcp-ovn-techpreview

**Pull request merged**